mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/MockWebServer.java

```diff
@@ -275,14 +277,18 @@ public void play() throws IOException {
    */
   public void play(int port) throws IOException {
     if (executor != null) throw new IllegalStateException("play() already called");
-    executor = Executors.newCachedThreadPool(Util.threadFactory("MockWebServer", false));
     inetAddress = InetAddress.getByName(null);
     serverSocket = serverSocketFactory.createServerSocket();
     serverSocket.setReuseAddress(port != 0); // Reuse the port if the port number was specified.
     serverSocket.bind(new InetSocketAddress(inetAddress, port), 50);
 
     this.port = serverSocket.getLocalPort();
-    executor.execute(new NamedRunnable("MockWebServer %s", this.port) {
+    this.service = sslSocketFactory != null ? protocols + ":" : "" + this.port;
```

Comment:
I was parsing reading this as

```
  sslSocketFactory != null ? (protocols + ":") : ("" + this.port);
```

instead of
```
 (sslSocketFactory != null ? protocols + ":" : "") + this.port;
```